docs/guides/code_push_add_to_app.md

```diff
@@ -0,0 +1,131 @@
+---
+sidebar_position: 4
+title: ➕ Code Push Add to App
+description: Use code push in an add-to-app scenario
+---
+
+# Code Push With Add-to-App
+
+This guide explains how to use Shorebird in an add-to-app scenario.
+
+## Prerequisites
+
+This guide assumes you have already have an Android app and a Flutter module. Our Android app will be named `android_app` and our Flutter module will be named `flutter_module`.
+
+This guide also assumes that you have created a Shorebird account and subscribed to Shorebird. If you have not, please see our [code push guide](../code-push) for instructions.
+
+The reference code for this guide is available at https://github.com/shorebirdtech/samples/tree/main/add_to_app.
+
+## Add Shorebird to your Flutter module
+
+First, run `shorebird init` in your Flutter module:
+
+```sh
+shorebird init
+```
+
+## Build your Flutter module with Shorebird
+
+Start by building your Flutter module as an Android archive (.aar):
+
+```sh
+shorebird build aar --build-number 1.0
+```
+
+:::note
+Because Shorebird only works with release builds, this will only produce a
+release version of your archive. This is similar to running
+`flutter build aar --no-debug --no-profile`.
+:::
+
+## Update your Android app to use the `download.shorebird.dev` Maven repository
+
+In `settings.gradle`:
+
+```diff
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        mavenCentral()
+        maven {
+            url '../my_flutter_module/build/host/outputs/repo'
+        }
++       maven {
+-           url 'https://storage.googleapis.com/download.flutter.io'
++           url 'https://download.shorebird.dev/download.flutter.io'
++      }
+    }
+}
+```
+
+:::note
+Even though we are replacing https://storage.googleapis.com/download.flutter.io.
+with https://download.shorebird.dev/download.flutter.io, any Flutter
+dependencies that are not unique to Shorebird will still be downloaded from
+https://storage.googleapis.com/download.flutter.io.
+:::
+
+## Update your Android app to use this version of the Flutter module
+
+In `app/build.gradle`, add the following:
+
+```diff
+dependencies {
+    // ...
++    releaseImplementation 'com.example.my_flutter_module:flutter_release:1.0'
+    // ...
+}
+```
+
+## Verify that your app runs
+
+In Android Studio, update the active build variant to release and run your app.
+Your app should work as before with no differences.
+
+## Release your Flutter module
+
+Create a Shorebird release for your Flutter module:
+
+```sh
+shorebird release aar --build-number 1.0 --release-version 1.2.3+1
```

Comment:
Isn't this redundant? Doesn't `release-version 1.2.3+1 imply `build-number` is 1? What happens if you run with:
```
shorebird release aar --build-number 1.0 --release-version 1.2.3+2
```

Reply:
If release-version contains build-number why is it necessary to also specify build-number explicitly? My concern is more that this provides an opportunity for a user to input conflicting information (`--build-number 1 --release-version 1.0.0+42`). Imo we should derive values rather than prompting for values whenever possible to reduce confusion/error

Reply:
The two numbers are completely unrelated. `release-version` is the app's release version, `build-number` is the module's build number. `--build-number 12.34 --release-version 5.6.7+890` is completely valid.

Reply:
To expand, `build-number` is what is used when importing the module in `app/build.grade`, as so:

```groovy
dependencies {
    // ...
   releaseImplementation 'com.example.my_flutter_module:flutter_release:12.34'
    // ...
}
```

And `release-version` is what we use elsewhere:

```groovy
    defaultConfig {
        applicationId "com.example.shorebirdaddtoapp"
        minSdk 24
        targetSdk 33
        versionCode 890
        versionName "5.6.7"
    }
```